### PR TITLE
expression: fix #4884, insert NULL on duplicate key by values function

### LINKED ISSUE #4884: insert NULL on duplicate key update val bug.
Please answer these questions before submitting your issue. Thanks!

```
drop table if exists test;
create table test(id int not null, val text, primary key(id));
insert into test values(1,'hello');
insert into test values(1, NULL) on duplicate key update val = VALUES(val);
select * from test; 
-- WRONG ANSWER: 1, ''
-- EXPECTED: 1, NULL 

select tidb_version();

Release Version: 1.0.0
Git Commit Hash: 1a175b562bc679694f606c336131a9373064b324
Git Branch: release-1.0
UTC Build Time: 2017-10-19 02:38:02
```

### PR DESCRIPTION
**Pull request by winkyao**
fix #4884 

@XuHuaiyu @zz-jason PTAL

### CODE REVIEW

File: expression/builtin_other.go [hunk 1]
```diff
@@ -361,12 +364,12 @@ func (b *builtinBitCountSig) evalInt(row types.Row) (int64, bool, error) {
...
 }
 
-// for plan cache of prepared statements
+// getParamFunctionClass for plan cache of prepared statements
 type getParamFunctionClass struct {
 	baseFunctionClass
 }
 
-//TODO: more typed functions will be added when typed parameters are supported.
+//TODO: getFunction more typed functions will be added when typed parameters are supported.
```
Comment (zimulala) [inline expression/builtin_other.go]: Add a blank after `//`

Comment (winkyao) [reply expression/builtin_other.go]: add blank will failed the pre-commit script by:

```
expression/builtin_other.go.372: // TODO: getFunction more typed functions will be added when typed parameters are supported. (godoc-local: TODO: -> getFunction?)
```

Comment (zimulala) [reply expression/builtin_other.go]:
```
// getFunction ....
// TODO: ....
```

**Comment by winkyao** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**